Components

RLewis627 opened the pull request:
Added all react components (src/components)
Added user-defined CSS (src/css)
Deleted items that are no longer needed (logos, icons, react-native CSS)
Added build folder with contents of `npm build`
Changed App.tsx to accommodate a multi-page web app with HashRouter

RLewis627 committed:
Delete package.json

RLewis627 committed:
Replaced deleted package.json

RLewis627 committed:
Delete index.css

Being replaced with user-defined CSS

RLewis627 committed:
Delete App.css

Being replace with user-defined CSS

RLewis627 committed:
Delete App.tsx

RLewis627 committed:
Delete logo.svg

Icons/logos no longer needed

RLewis627 committed:
Added new App.tsx

RLewis627 committed:
Added MainStyle.css

RLewis627 committed:
Rename src/MainStyle.css to src/css/MainStyle.css

RLewis627 committed:
Added rest of CSS files

RLewis627 committed:
Added Conversation component

RLewis627 committed:
Rename src/Conversation.js to src/components/Conversation.js

RLewis627 committed:
Added other components

Login component
Thank You page
Registration component

RLewis627 committed:
Delete favicon.ico

Icons/logos no longer needed

RLewis627 committed:
Delete logo192.png

Icons/logos no longer needed

RLewis627 committed:
Delete logo512.png

Icons/logos no longer needed

RLewis627 committed:
Update index.html

RLewis627 committed:
Update manifest.json

Icons/logos no longer needed

RLewis627 committed:
Update index.tsx

RLewis627 committed:
Add files via upload

RLewis627 committed:
Rename asset-manifest.json to build/asset-manifest.json

RLewis627 committed:
Add files via upload

RLewis627 committed:
Add files via upload

RLewis627 committed:
Rename build/main.fa28235b.chunk.css to build/static/ccss/main.fa28235b.chunk.css

RLewis627 committed:
Delete main.fa28235b.chunk.css

RLewis627 committed:
Add files via upload

RLewis627 committed:
Rename build/main.fa28235b.chunk.css to build/static/css/main.fa28235b.chunk.css

RLewis627 committed:
Add files via upload

RLewis627 committed:
Add files via upload

RLewis627 committed:
Rename build/static/2.bfcd2794.chunk.js.LICENSE.txt to build/static/js/2.bfcd2794.chunk.js.LICENSE.txt

RLewis627 committed:
Add files via upload

Nicholas-Baron commented on src/components/Thanks.js:
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import '../css/ThankYouPage.css';
+
+class Thanks extends React.Component {
+  render() {
+    return <p>Thank you for using Therabot!</p>;
+  }
+}
```
Could this be rewritten as a function?
```
export default function Thanks(){
  return ... ;
}
```

Nicholas-Baron commented on src/components/UserLoginBox.js:
```diff
@@ -0,0 +1,50 @@
+  }
+
+  ChkSignIn() {
+      var LoginUseName = document.getElementById("LoginUseName").value;
+      var Loginpin = document.getElementById("LoginPIN").value;
```
Could these be `let` or even `const`?

Nicholas-Baron commented on src/components/Register.js:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    ChkSuccessReg() {
+        var FirstName = document.getElementById("FirstName").value;
+        var UseName = document.getElementById("UseName").value;
+        var EmailAdd = document.getElementById("EmailAdd").value;
+        var pin = document.getElementById("pin").value;
```
Could these be `let` or `const`?

Nicholas-Baron commented on src/components/Conversation.js:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    AppendUserConvo() {
+        var UserComment = document.getElementById("ConvoUserInput").value;
```
Could this be `let` or `const`?

Nicholas-Baron commented on package.json:
```diff
@@ -11,11 +11,17 @@
+    "react-router": "^5.1.2",
+    "react-router-dom": "^5.1.2",
     "react-scripts": "3.3.1",
+    "reactstrap": "^8.4.1",
```
I'll allow it, but please limit the usage of this package.

Nicholas-Baron reviewed (changes requested):
I strongly recommend removing the build folder from this commit. It is unnecessary and just bloats the repository.
When the pull request merges, I will begin writing the tests in Typescript and will convert the files to Typescript.

RLewis627 committed:
Delete Thanks.js

Converted it to a function

RLewis627 committed:
Update Conversation.js

Converted ThankYou component into a function that is integrated into the Conversation component
Added POST request method to firebase server

RLewis627 committed:
Update Register.js

Changed all vars to const

RLewis627 committed:
Update UserLoginBox.js

Changed all vars to const

RLewis627 committed:
Update App.tsx

Thanks component no longer exists

RLewis627 committed:
Update package.json

Removed reactstrap dependency.

RLewis627 committed:
Delete service-worker.js

RLewis627 committed:
Delete robots.txt

RLewis627 committed:
Delete precache-manifest.c2a16b11c9acfb2dc10fdaff9ce77a3b.js

RLewis627 committed:
Delete manifest.json

RLewis627 committed:
Delete index.html

RLewis627 committed:
Delete asset-manifest.json

RLewis627 committed:
Delete main.fa28235b.chunk.css.map

RLewis627 committed:
Delete main.fa28235b.chunk.css

RLewis627 committed:
Delete runtime-main.1c5d4cde.js.map

RLewis627 committed:
Delete runtime-main.1c5d4cde.js

RLewis627 committed:
Delete main.9532c921.chunk.js.map

RLewis627 committed:
Delete main.9532c921.chunk.js

RLewis627 committed:
Delete 2.bfcd2794.chunk.js.map

RLewis627 committed:
Delete 2.bfcd2794.chunk.js.LICENSE.txt

RLewis627 committed:
Delete 2.bfcd2794.chunk.js

RLewis627 commented:
- Also removed build folder
- Attempted to remove jquery dependency as well, but converting the 2 Jquery functions that exist in Conversation.js to their Javascript counterparts resolves to various errors in the console.
- Attempted to convert files to tsx but it resolves to a large amount of errors that I don't really know how to fix.

Nicholas-Baron commented on src/components/Conversation.js:
```diff
@@ -0,0 +1,93 @@
+//import firebase from "../firebase.js"
+
+class Conversation extends React.Component {
+    constructor(props) {
+        super(props);
```
The props are unused. I will rewrite the class as a functional component.

Nicholas-Baron reviewed (approved):
I will begin rewriting the class style components into function style components and adding tests.
